fpga_interchange/yosys_json.py

```diff
@@ -18,20 +18,44 @@
         CellInstance, Direction, Library
 from fpga_interchange.parameter_definitions import ParameterFormat
 
+BUS_INDEX_RE = re.compile(r"(?P<name>[^\s\[\]]*)(\[(?P<index>[0-9]+)\])?")
 
-def is_bus(bits, offset, upto):
+
+def is_bus(name, bits, offset, upto):
     """ Returns true if this port/net is a bus.
 
-    >>> is_bus([2], offset=0, upto=0)
+    >>> is_bus("net", [2], offset=0, upto=0)
     False
-    >>> is_bus([2, 3], offset=0, upto=0)
+    >>> is_bus("net", [2, 3], offset=0, upto=0)
     True
-    >>> is_bus([2], offset=1, upto=0)
+    >>> is_bus("net", [2], offset=1, upto=0)
     True
-    >>> is_bus([2], offset=0, upto=1)
+    >>> is_bus("net", [2], offset=0, upto=1)
     True
+    >>> is_bus("net[3]", [100], offset=3, upto=0)
+    False
 
```

Comment:
Should probably add a test for the following case:
```
is_bus("net[3]", [100], offset=4, upto=0)
True
```

As far as I understand from the below comment, if the offset and index in the brackets are not the same, the net should be considered as a bus

Reply:
I added the test case.